packages/grafana-ui/src/utils/singlestat.ts

```diff
@@ -1,11 +1,17 @@
 import { PanelData, NullValueMode, SingleStatValueInfo } from '../types';
 import { processTimeSeries } from './processTimeSeries';
+import { DisplayValueOptions } from './displayValue';
 
 export interface SingleStatProcessingOptions {
   panelData: PanelData;
   stat: string;
 }
 
+export interface SingleStatOptions {
+  stat: string;
+  display: DisplayValueOptions;
```

Comment:
is there a reason to merge everything here? Think the old props makes more sense (valueOptions, thresholds, have stat be part of valueOptions). Display options I want to be unique to each visualization, BarGauge will need some display options (mode, lcd/simple/gradient), cell count etc.

Reply:
my big goal with this is to separate *stat* from all the display stuff.  'stat' is really the source of the data where everything else is about what to display given the data.

stat, should easily be replaced with column reference, a function, etc (but I'm not touching that now!)

Yes every panel should have special props, but many of them are common and used everywhere: unit, mappings, threshold, prefix, etc.  Currently each panel deals with routing them together.  My goal was to put the common ones together and have a shred utility that deals with them.

This only deals with mapping the source data->{string,number} all the other details are up to the panel.